clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java

```diff
@@ -281,64 +276,15 @@ void testEnsureMetadataUpdateOnPoll() {
     }
 
     @Test
-    void testEnsureEventsAreCompleted() {
```

Comment:
Why did you remove this test without replacement?

Reply:
Reinstated.

Reply:
Actually seems to me that we shouldn't have this test here (and maybe this is why @kirktrue removed it before?). As I see it, this unit test is testing something that is not the `ConsumerNetworkThread`'s responsibility (and that's why it ends up being complicate, having to mimic the reaper behaviour and spying). It is testing that events are completed, and that's the reaper.reap responsibility, so seems to me we need to:

1. test that the `ConsumerNetworkThread` calls the reaper with the full list of events -> done already in the [testCleanupInvokesReaper](https://github.com/apache/kafka/blob/91af164415b8b950c70d3a61bb0837c34ae4ed69/clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThreadTest.java#L331)
2. test that the `CompletableEventReaper.reap(Collection<?> events)` completes the events, and that's done in CompletableEventReaperTest ([testIncompleteQueue](https://github.com/apache/kafka/blob/91af164415b8b950c70d3a61bb0837c34ae4ed69/clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/CompletableEventReaperTest.java#L135) and [testIncompleteTracked](https://github.com/apache/kafka/blob/91af164415b8b950c70d3a61bb0837c34ae4ed69/clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/CompletableEventReaperTest.java#L170)) 

In the end, as it is, we end up asserting a behaviour we're mocking ourselves in the `doAnswer`, so not much value I would say? Agree with @cadonna that we need coverage, but I would say that's what we have on my points 1 and 2, and this should be removed. Makes sense?

Reply:
Yes, the test was a little suspect in terms of its value-add, so I'd removed it.

I was planning to file a Jira to move several of the tests (including this one) from `ConsumerNetworkThreadTest` to `ApplicationEventProcessorTest`. Then we could fix up some of the funkiness in this test as a separate task.

Reply:
That is all fine! I was not arguing that we need to keep the test, but if I see a test removed without replacement, I suspect a mistake. Which did apparently not happen in this case. Next time comment on the PR why you removed the test.